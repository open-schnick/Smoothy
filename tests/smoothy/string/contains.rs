use crate::failing_assertion;
use smoothy::{assert_that, StringAssertion};

#[allow(clippy::unnecessary_to_owned)]
#[test]
fn succeeds_with_str() {
    assert_that("Hello World")
        .contains("Hello")
        .and()
        .contains("World".to_string());
}

#[allow(clippy::unnecessary_to_owned)]
#[test]
fn succeeds_with_string() {
    assert_that("Hello World".to_string())
        .contains("Hello")
        .and()
        .contains("World".to_string());
}

#[test]
fn fails() {
    failing_assertion!({
        assert_that("Hello World").contains("BlaFasel");
    });
}
