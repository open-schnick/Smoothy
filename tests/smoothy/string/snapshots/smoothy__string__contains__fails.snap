---
source: tests/smoothy/string/contains.rs
expression: assertion_failed_output
---
[31mAssertion failed![0m

Expected
  [31m"Hello World"[0m
to contain
  [32m"BlaFasel"[0m
